mantid/hardware-requirements/plan.md

```diff
@@ -0,0 +1,129 @@
+
+- Due to the interactive nature, high CPU use will alternate with shorter or longer idle breaks.
+  However, generally there will be a rather large permanent memory requirement, i.e., clustering interactive user sessions on a single compute node is only possible to a limited extent, and will require nodes with a large amount of memory.
+
+### Script-based reduction
+
+Reduction of data based on a (Python) script.
+This covers auto reduction and batch reduction.
+
+## Prerequisites
+
+### Baseline
+
+To establish a baseline, we need to survey the hardware used for data reduction at existing facilities using Mantid.
+This should include computers at beamlines, laptops and desktop PCs used by instrument scientists and users, and cluster hardware resources.
+
+### From instrument teams
+
+- A similar reduction workflow (in Mantid) at an operating facility (should be ISIS or SNS) that we can use for benchmarking.
+  - Could be several if the instrument has different operating modes.
+- Listing of build-out phases including, for each phase:
+  - Detector pixel counts.
+  - Estimated typical event rates (should be given depending on accelerator power).
+  - Typical length of a single run (single measurement for a given sample).
+    Could be given as wall clock time or as event count.
+- Other:
+  - Are there large event-mode monitors?
+    This is known to have caused issues at SNS.
+
+## Result presentation
+
+The results will depend on a number of variables.
+Presentation as a spreadsheet is probably most useful, but auxiliary documentation is required.
+The compute need should be given for individual instruments.
+This will give us flexibility when the schedule changes.
+
+### Required information
+
+Most likely a heterogeneous pool of hardware suits our needs best.
+In particular, the optimal balance between compute power and main memory will be different for interactive reduction and script-based reduction.
+The main focus of the results should be:
+
+- Number of required cores.
+- Amount of main memory.
+- Disk I/O.
+
+Furthermore:
+
+- Network.
+  Likely not an issue, but  Ethernet is not sufficient in general.
+  Infiniband or equivalent will do, even if not top-of-the-line.
+- Some workflows may benefit from fast/large local SSD storage, in particular file-backed operation for working with `MDEventWorkspace`.
+  It is currently not clear to what extent this would be used.
+
+For all cases, we need to determine the requirements at peak times and averages.
+For example:
+
+- All live reduction must be available simultaneously.
+- Interactive reduction peaks during day times, and during and soon after phases with beam on.
+  This may be important during hot commissioning where we probably have short phases of beam with longer breaks?
+- Do we need to "guarantee" a turnaround time for script-based reduction?
```

Comment:
I think this would be hard if not impossible to do. I would steer clear of any guarantees.

Reply:
I would think we do, at least for the users who are currently running an experiment. We cannot have a submission queue in the cluster where their jobs are waiting longer than their experiment takes.

With proper settings for maximum job runtimes and job prioritization this should not be too hard to achive. Note that this is not the same as saying that *all* users will have a guaranteed turnaround time for *all* their jobs.

Reply:
Ah ok. You are talking about the waiting time in the queue, I thought you were talking about the execution time of the scripts themselves.